pkg/crinit/util/util.go

```diff
@@ -53,6 +158,788 @@ func (o *SubcommandOptions) SetName(args []string) error {
+	} else {
+		o.ApiServerNodePortPortPtr = nil
+	}
+
+	if o.ApiServerNodePortPort < 0 || o.ApiServerNodePortPort > 65535 {
+		return fmt.Errorf("Please provide a valid port number for %s", apiserverPortFlag)
+	}
+
+	return nil
+}
+
+// marshalOptions marshals options if necessary.
+func (o *SubcommandOptions) MarshalOptions() error {
+	if o.ApiServerOverridesString == "" {
+		return nil
+	}
+
+	argsMap := make(map[string]string)
+	overrideArgs := strings.Split(o.ApiServerOverridesString, ",")
+	for _, overrideArg := range overrideArgs {
+		splitArg := strings.SplitN(overrideArg, "=", 2)
+		if len(splitArg) != 2 {
+			return fmt.Errorf("wrong format for override arg: %s", overrideArg)
+		}
+		key := strings.TrimSpace(splitArg[0])
+		val := strings.TrimSpace(splitArg[1])
+		if len(key) == 0 {
+			return fmt.Errorf("wrong format for override arg: %s, arg name cannot be empty", overrideArg)
+		}
+		argsMap[key] = val
+	}
+
+	o.ApiServerOverrides = argsMap
+
+	return nil
+}
+
+// CreateNamespace creates the cluster registry namespace
+func (o *SubcommandOptions) CreateNamespace(cmdOut io.Writer,
+	clientset client.Interface) error {
+
+	fmt.Fprintf(cmdOut, "Creating a namespace %s for the cluster registry...",
+		o.ClusterRegistryNamespace)
+	glog.V(4).Infof("Creating a namespace %s for the cluster registry",
+		o.ClusterRegistryNamespace)
+
+	_, err := CreateNamespace(clientset, o.ClusterRegistryNamespace, o.DryRun)
+
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintln(cmdOut, " done")
+	return err
+}
+
+// CreateNamespace helper to create the cluster registry namespace object and return
+// the object
+func CreateNamespace(clientset client.Interface, namespace string,
+	dryRun bool) (*v1.Namespace, error) {
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+		},
+	}
+
+	if dryRun {
+		return ns, nil
+	}
+
+	return clientset.CoreV1().Namespaces().Create(ns)
+}
+
+// CreateService creates the cluster registry apiserver service
+func (o *SubcommandOptions) CreateService(cmdOut io.Writer,
+	clientset client.Interface) (*v1.Service, []string, []string, error) {
+
+	fmt.Fprint(cmdOut, "Creating cluster registry API server service...")
+	glog.V(4).Info("Creating cluster registry API server service")
+
+	svc, ips, hostnames, err := CreateService(cmdOut, clientset,
+		o.ClusterRegistryNamespace, o.Name, o.ApiServerAdvertiseAddress,
+		o.ApiServerNodePortPortPtr, o.ApiServerServiceType, o.DryRun)
+
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	fmt.Fprintln(cmdOut, " done")
+	glog.V(4).Infof("Created service named %s with IP addresses %v, hostnames %v",
+		svc.Name, ips, hostnames)
+
+	return svc, ips, hostnames, err
+}
+
+// CreateService helper to create the cluster registry apiserver service object
+// and return the object. If service type is load balancer, will wait for load
+// balancer IP and return it and hostnames
+func CreateService(cmdOut io.Writer, clientset client.Interface, namespace,
```

Comment:
Should this be private?

Reply:
It is used in `standalone_test.go`.

Reply:
Ah, I see. I assume that will change in the follow-up, when the standalone tests are moved to util.